include/dxc/DXIL/DxilTypeSystem.h

```diff
@@ -139,6 +139,48 @@ class DxilStructAnnotation {
   std::vector<DxilTemplateArgAnnotation> m_TemplateAnnotations;
 };
 
+enum class PayloadAccessTypes{ ReadWrite = 0, Read = 1, Write = 2 };
```

Comment:
I renamed the type and moved it to DXILConstants with a bunch of other improvements. 

A dev would declare are field as padding with the full set of qualifiers like: 
`int x : read(caller, closesthit, miss, anyhit) : write(caller, closesthit, miss, anyhit);`
which basically replicates the current behavior.

Reply:
I was trying to express the opposite intent.  It certainly was a hypothetical, since I was trying to think of a use case for a field that had no read or write expressed for any stage.  The idea was maybe one wanted to pad the structure for cases where the structure is used as-is, but not consume space for payloads optimized by the drivers using these qualifiers.

Reply:
We decided to allow empty payload qualifiers like `int x : read() : write();` this should enable padding like you mentioned.